services/register.js

```diff
@@ -0,0 +1,42 @@
+const userRepository = require("../repositories/user")
+const bcrypt = require("bcrypt")
+const errors = require("../configuration/errors")
+const userRoles = require("../configuration/user-roles")
+const passwordValidator = require("../validators/password")
+
+
+async function registerUser(email, password, role) {
```

Comment:
also we can take here 1 parameter "user". and that parameter will have {email:dlsakfldkasf, password:laksdfjlasfj, role:USER}. I think that is req.body. 
and then you can use it like user.email, user.password...